week2/unit_model_configuration_verbose.ipynb

```diff
@@ -0,0 +1,909 @@
+    "\n",
+    "Passing different configuration arguments will typically result in the creation (or exclusion) of specific variables and constraints.\n",
+    "\n",
+    "Some models have many configuration arguments and some have zero. All WaterTAP unit process models require passing the required property model via the `property_package` configuration argument. This tutorial will go through configuration arguments for the WaterTAP pump and RO model.\n",
+    "\n",
+    "<center><img src=\"graphics/config_arguments.png\" width=\"700\" /></center>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d1a75d0e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "# Imports from Pyomo\n",
+    "from pyomo.environ import (\n",
+    "    ConcreteModel,\n",
+    "    value,\n",
+    "    assert_optimal_termination,\n",
+    "    units as pyunits,\n",
+    ")\n",
+    "\n",
+    "# Imports from IDAES\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from idaes.core.util.scaling import calculate_scaling_factors, set_scaling_factor\n",
+    "\n",
+    "# Imports from WaterTAP\n",
+    "from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock\n",
+    "from watertap.unit_models.pressure_changer import Pump\n",
+    "from watertap.unit_models.reverse_osmosis_0D import ReverseOsmosis0D\n",
+    "from watertap.core.solvers import get_solver\n",
+    "\n",
+    "solver = get_solver()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "edcf040f",
+   "metadata": {},
+   "source": [
+    "# Pump Configuration Arguments\n",
+    "\n",
+    "Every WaterTAP model has default values for every configuration argument (except `property_package`). Using different model configurations will create additional variables and constraints that require additional parameter data from the user. \n",
+    "\n",
+    "The example below presents two different builds for the WaterTAP `Pump` model: the first uses the default value for the `variable_efficiency` argument (`\"none\"`) while the second uses the `\"flow\"` argument. The second build creates three additional variables (`bep_flow`, `bep_eta`, and `flow_ratio`) and one additional constraints (`flow_ratio_constraint`)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "97415603",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Pump parameters\n",
+    "eta = 0.8\n",
+    "bep_flow = 1e-3\n",
+    "salinity = 35  # g/L\n",
+    "temperature = 298  # K\n",
+    "inlet_pressure = 40 * pyunits.bar\n",
+    "deltaP = 10 * pyunits.bar\n",
+    "\n",
+    "flows = np.linspace(1e-4, bep_flow, 25)  # m3/s"
```

Comment:
```suggestion
    "flows = np.linspace(6e-4, bep_flow, 25)  # m3/s"
```

Reply:
Let's cut off the low region just because this is a default relationship and not best representation of typical variable efficiency observed (sharp discontinuity). It should be enough to demonstrate a continuous and less pronounced shift in efficiency and power consumption, with the message that we can add new relationships as desired.